velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +367,64 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar, validPattern_)),
+            RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    // apply() will not be invoked if the pattern was invalid or the selection
```

Comment:
Is it possible to have an invalid regexp pattern here? If it is, wouldn't it be a programming error? If so, we should not throw an USER error.

Reply:
You are right that invalid pattern should be reported first. Invalid pattern didn't actually mean bad re.

I think the latest code is cleanest logic. ptal.